{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/bmigliozzi/anaconda/lib/python2.7/site-packages/matplotlib/font_manager.py:273: UserWarning: Matplotlib is building the font cache using fc-list. This may take a moment.\n",
      "  warnings.warn('Matplotlib is building the font cache using fc-list. This may take a moment.')\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "from IPython.display import display\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "sns.set_context(\"notebook\", font_scale=1)\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline \n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hdAct1 = 'AAAAGACATCTTTTCCAAATCAGTCAGTTCCAGCACCGACGCCACCCGGATTCCCCCGTCATCGCATCGCTGTTCGCTATTCATCCCTGCGCTATTTTTCACTTTAGTTTCGCGTAATCCCTCAGAACAGTCGCAATGGAAGACGAAGTTGCCGCCTTGGTCGTGGACAATGGATCCGGTATGTGCAAGGCCGGATTTGCCGGAGATGACGCTCCCCGCGCCGTCTTCCCCTCCATCGTTGGCCGACCCCGTCATCAGGGTGTCATGGTCGGTATGGGTCAAAAGGACAGCTACGTCGGTGATGAGGCCCAGAGCAAGCGCGGTATCCTGACGCTCAAGTACCCCATCGAGCACGGCATCGTCACCAACTGGGATGACATGGAGAAGATCTGGCATCACACCTTCTACAACGAGCTCCGCGTGGCTCCCGAGGAACACCCCGTCCTCCTGACTGAGGCTCCCCTCAACCCCAAGGCCAACAGGGAAAAGATGACCCAGATCATGTTCGAGACATTCAACACCCCCGCCATGTACGTCGCGATTCAGGCTGTGCTCTCCCTGTACGCGTCCGGTCGTACCACCGGTATCGTGCTGGACTCTGGTGATGGTGTCTCCCACACTGTCCCCATCTACGAAGGTTATGCTCTGCCTCACGCCATCCTCCGTCTGGATTTGGCCGGTCGCGACTTGACTGACTACTTGATGAAGATCCTGACTGAGCGCGGTACAGCTTCGTCACAACGCT'\n",
    "hdMag1 = 'CCCATTAGAGGGATTTCATCTATTTCAACTCTTGGGCATTTTCTCTGACGCTTAAATTTTTCCCAGACAATCTCGAAGGAAAACGAAAATGACGGAGGACCGTTTCTACGTGCGATACTACGTCGGCCATAAAGGCAAGTTCGGGCATGAATTTTTGGAATTCGAATTTCGCCCGGACGGTCGCCTGCGTTACGCCAACAATTCCAACTACAAGAACGACACGATGATCCGCAAGGAGGTGGTCGTCCATCCGGCTATCTTGGAAGAGGTGAAGCGCATCATTCAGGACAGTGAGATTCTCCGGGAGAACGACGCCAAGTGGCCGCAGCCGGACCGCGTGGGCCGGCAGGAGTTAGAGATTCTGCTGGACGACGAGCACATCTCCTTCAACACGGGAAAGATCGGCTCGCTGATGGACGTCAACAACAGCCCCGATCCAGAAGGTCTGCGTTGCTTCTACTACCTCGTGCAGGATCTCAAGTGTCTCGTCTTCTCCCTCATTGCGCTGCACTTTAAAATCAAGCCGATTTGAACGCTTGTAATCAACCAATCAAGCGCGAAGATTTGCACTAGCACCCTCGTCGAC'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 302 ms, sys: 48.7 ms, total: 351 ms\n",
      "Wall time: 584 ms\n"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "df =  pd.read_csv('pam_sites.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>nHd.2.3.scaf00001</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>GGTGACTGTATTCCGGGGGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>nHd.2.3.scaf00001</td>\n",
       "      <td>2875</td>\n",
       "      <td>2894</td>\n",
       "      <td>GAATGCATTTCGTTGAATGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nHd.2.3.scaf00001</td>\n",
       "      <td>3222</td>\n",
       "      <td>3241</td>\n",
       "      <td>GAATGCATTTCGCTGAATGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nHd.2.3.scaf00001</td>\n",
       "      <td>3291</td>\n",
       "      <td>3310</td>\n",
       "      <td>GCGGGCATCTCATTGTCGGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>nHd.2.3.scaf00001</td>\n",
       "      <td>3594</td>\n",
       "      <td>3613</td>\n",
       "      <td>GCATGCGCTGTTTCCAACGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Contig  Start   End              gRNA_Seq  PAM Strand G_start  \\\n",
       "0  nHd.2.3.scaf00001      2    21  GGTGACTGTATTCCGGGGGG  AGG      +    True   \n",
       "1  nHd.2.3.scaf00001   2875  2894  GAATGCATTTCGTTGAATGG  AGG      +    True   \n",
       "2  nHd.2.3.scaf00001   3222  3241  GAATGCATTTCGCTGAATGG  AGG      +    True   \n",
       "3  nHd.2.3.scaf00001   3291  3310  GCGGGCATCTCATTGTCGGG  AGG      +    True   \n",
       "4  nHd.2.3.scaf00001   3594  3613  GCATGCGCTGTTTCCAACGG  AGG      +    True   \n",
       "\n",
       "  Unique  \n",
       "0     No  \n",
       "1     No  \n",
       "2     No  \n",
       "3    Yes  \n",
       "4     No  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td>239950</td>\n",
       "      <td>38268.9</td>\n",
       "      <td>51313.5</td>\n",
       "      <td>1</td>\n",
       "      <td>6532</td>\n",
       "      <td>20655.5</td>\n",
       "      <td>49273.8</td>\n",
       "      <td>594043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td>239950</td>\n",
       "      <td>38268.9</td>\n",
       "      <td>51313.5</td>\n",
       "      <td>4</td>\n",
       "      <td>6530</td>\n",
       "      <td>20660.5</td>\n",
       "      <td>49270.8</td>\n",
       "      <td>594024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G_start</th>\n",
       "      <td>239950</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          count     mean      std   min   25%      50%      75%     max\n",
       "Start    239950  38268.9  51313.5     1  6532  20655.5  49273.8  594043\n",
       "End      239950  38268.9  51313.5     4  6530  20660.5  49270.8  594024\n",
       "G_start  239950        1        0  True     1        1        1    True"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def pairDNA(x):\n",
    "    if(x=='A'):\n",
    "        return 'T'\n",
    "    elif(x=='T'):\n",
    "        return 'A'\n",
    "    elif(x=='C'):\n",
    "        return 'G'\n",
    "    else:\n",
    "        return 'C'\n",
    "\n",
    "def dnaComp(s):\n",
    "    # Gets complementary strand of DNA\n",
    "    return ''.join(map(lambda b: pairDNA(b), s))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "df['gRNA_Seq_Comp'] = map(lambda x: dnaComp(x), df['gRNA_Seq'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td>223</td>\n",
       "      <td>50446.3</td>\n",
       "      <td>28581.9</td>\n",
       "      <td>1953</td>\n",
       "      <td>25853</td>\n",
       "      <td>56215</td>\n",
       "      <td>73148</td>\n",
       "      <td>100028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td>223</td>\n",
       "      <td>50445.6</td>\n",
       "      <td>28583</td>\n",
       "      <td>1934</td>\n",
       "      <td>25853</td>\n",
       "      <td>56234</td>\n",
       "      <td>73129</td>\n",
       "      <td>100047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G_start</th>\n",
       "      <td>223</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        count     mean      std   min    25%    50%    75%     max\n",
       "Start     223  50446.3  28581.9  1953  25853  56215  73148  100028\n",
       "End       223  50445.6    28583  1934  25853  56234  73129  100047\n",
       "G_start   223        1        0  True      1      1      1    True"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td>575</td>\n",
       "      <td>130381</td>\n",
       "      <td>70488.5</td>\n",
       "      <td>593</td>\n",
       "      <td>74255</td>\n",
       "      <td>138397</td>\n",
       "      <td>192537</td>\n",
       "      <td>252622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td>575</td>\n",
       "      <td>130382</td>\n",
       "      <td>70490.4</td>\n",
       "      <td>574</td>\n",
       "      <td>74255</td>\n",
       "      <td>138416</td>\n",
       "      <td>192518</td>\n",
       "      <td>252603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G_start</th>\n",
       "      <td>575</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        count    mean      std   min    25%     50%     75%     max\n",
       "Start     575  130381  70488.5   593  74255  138397  192537  252622\n",
       "End       575  130382  70490.4   574  74255  138416  192518  252603\n",
       "G_start   575       1        0  True      1       1       1    True"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "scaf209_df = df[df.Contig=='nHd.2.3.scaf00209']\n",
    "display(scaf209_df.describe().T)\n",
    "scaf20_df = df[df.Contig=='nHd.2.3.scaf00020']\n",
    "display(scaf20_df.describe().T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "      <th>gRNA_Seq_Comp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60501</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>44951</td>\n",
       "      <td>44970</td>\n",
       "      <td>GCTTGCTTAGTACCTCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CGAACGAATCATGGAGGGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60543</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>79241</td>\n",
       "      <td>79260</td>\n",
       "      <td>GAACAGTGGTGCCAGTAAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTGTCACCACGGTCATTCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60497</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>42928</td>\n",
       "      <td>42947</td>\n",
       "      <td>GTATGAAATTGCCAGAGTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CATACTTTAACGGTCTCACC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60553</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>79470</td>\n",
       "      <td>79489</td>\n",
       "      <td>GGAGGAGGAGGAGGAGTGGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CCTCCTCCTCCTCCTCACCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60661</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>78941</td>\n",
       "      <td>78922</td>\n",
       "      <td>GGTGGAGGCGAGGAGCTCGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCACCTCCGCTCCTCGAGCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Contig  Start    End              gRNA_Seq  PAM Strand  \\\n",
       "60501  nHd.2.3.scaf00209  44951  44970  GCTTGCTTAGTACCTCCCGG  CGG      +   \n",
       "60543  nHd.2.3.scaf00209  79241  79260  GAACAGTGGTGCCAGTAAGG  CGG      +   \n",
       "60497  nHd.2.3.scaf00209  42928  42947  GTATGAAATTGCCAGAGTGG  CGG      +   \n",
       "60553  nHd.2.3.scaf00209  79470  79489  GGAGGAGGAGGAGGAGTGGG  AGG      +   \n",
       "60661  nHd.2.3.scaf00209  78941  78922  GGTGGAGGCGAGGAGCTCGG  AGG      -   \n",
       "\n",
       "      G_start Unique         gRNA_Seq_Comp  \n",
       "60501    True     No  CGAACGAATCATGGAGGGCC  \n",
       "60543    True    Yes  CTTGTCACCACGGTCATTCC  \n",
       "60497    True    Yes  CATACTTTAACGGTCTCACC  \n",
       "60553    True     No  CCTCCTCCTCCTCCTCACCC  \n",
       "60661    True    Yes  CCACCTCCGCTCCTCGAGCC  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "      <th>gRNA_Seq_Comp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11553</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>138012</td>\n",
       "      <td>137993</td>\n",
       "      <td>GTTCGGTGAGGACGTCGAGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAAGCCACTCCTGCAGCTCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11144</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>45448</td>\n",
       "      <td>45467</td>\n",
       "      <td>GCATTTCAGGGTGACGCTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGTAAAGTCCCACTGCGACC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11246</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>143517</td>\n",
       "      <td>143536</td>\n",
       "      <td>GTCATGCGTCCTGCCCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAGTACGCAGGACGGGGGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11508</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>92023</td>\n",
       "      <td>92004</td>\n",
       "      <td>GCCCATTATGGATGACCCGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGGGTAATACCTACTGGGCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11509</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>92554</td>\n",
       "      <td>92535</td>\n",
       "      <td>GCATTACTAAGACGTATGGG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CGTAATGATTCTGCATACCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Contig   Start     End              gRNA_Seq  PAM Strand  \\\n",
       "11553  nHd.2.3.scaf00020  138012  137993  GTTCGGTGAGGACGTCGAGG  AGG      -   \n",
       "11144  nHd.2.3.scaf00020   45448   45467  GCATTTCAGGGTGACGCTGG  CGG      +   \n",
       "11246  nHd.2.3.scaf00020  143517  143536  GTCATGCGTCCTGCCCCCGG  CGG      +   \n",
       "11508  nHd.2.3.scaf00020   92023   92004  GCCCATTATGGATGACCCGG  AGG      -   \n",
       "11509  nHd.2.3.scaf00020   92554   92535  GCATTACTAAGACGTATGGG  TGG      -   \n",
       "\n",
       "      G_start Unique         gRNA_Seq_Comp  \n",
       "11553    True    Yes  CAAGCCACTCCTGCAGCTCC  \n",
       "11144    True    Yes  CGTAAAGTCCCACTGCGACC  \n",
       "11246    True    Yes  CAGTACGCAGGACGGGGGCC  \n",
       "11508    True    Yes  CGGGTAATACCTACTGGGCC  \n",
       "11509    True     No  CGTAATGATTCTGCATACCC  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(scaf209_df.sample(5))\n",
    "display(scaf20_df.sample(5))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def matchSeq(gRNA,gene):\n",
    "    match = re.findall(gRNA,gene)\n",
    "    if(len(match)>0):\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Total Matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "scaf209_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdAct1),scaf209_df['gRNA_Seq'])\n",
    "scaf20_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdMag1),scaf20_df['gRNA_Seq'])\n",
    "\n",
    "print(len(scaf209_df[scaf209_df['Match_gRNA']==True]))\n",
    "print(len(scaf20_df[scaf20_df['Match_gRNA']==True]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Unique Matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td>160</td>\n",
       "      <td>48657.5</td>\n",
       "      <td>28897.3</td>\n",
       "      <td>1953</td>\n",
       "      <td>23643.8</td>\n",
       "      <td>47753.5</td>\n",
       "      <td>73145.5</td>\n",
       "      <td>99529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td>160</td>\n",
       "      <td>48655.4</td>\n",
       "      <td>28895.9</td>\n",
       "      <td>1934</td>\n",
       "      <td>23624.8</td>\n",
       "      <td>47753.5</td>\n",
       "      <td>73126.5</td>\n",
       "      <td>99510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G_start</th>\n",
       "      <td>160</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        count     mean      std   min      25%      50%      75%    max\n",
       "Start     160  48657.5  28897.3  1953  23643.8  47753.5  73145.5  99529\n",
       "End       160  48655.4  28895.9  1934  23624.8  47753.5  73126.5  99510\n",
       "G_start   160        1        0  True        1        1        1   True"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Start</th>\n",
       "      <td>430</td>\n",
       "      <td>124202</td>\n",
       "      <td>69213</td>\n",
       "      <td>593</td>\n",
       "      <td>73514.8</td>\n",
       "      <td>118231</td>\n",
       "      <td>189411</td>\n",
       "      <td>252622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>End</th>\n",
       "      <td>430</td>\n",
       "      <td>124202</td>\n",
       "      <td>69214</td>\n",
       "      <td>574</td>\n",
       "      <td>73495.8</td>\n",
       "      <td>118250</td>\n",
       "      <td>189402</td>\n",
       "      <td>252603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G_start</th>\n",
       "      <td>430</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        count    mean    std   min      25%     50%     75%     max\n",
       "Start     430  124202  69213   593  73514.8  118231  189411  252622\n",
       "End       430  124202  69214   574  73495.8  118250  189402  252603\n",
       "G_start   430       1      0  True        1       1       1    True"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "uniq_df = df[df.Unique=='Yes']\n",
    "uniq_df.describe().T\n",
    "\n",
    "uniq_scaf209_df = uniq_df[uniq_df.Contig=='nHd.2.3.scaf00209']\n",
    "display(uniq_scaf209_df.describe().T)\n",
    "uniq_scaf20_df = uniq_df[uniq_df.Contig=='nHd.2.3.scaf00020']\n",
    "display(uniq_scaf20_df.describe().T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "uniq_scaf209_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdAct1),uniq_scaf209_df['gRNA_Seq'])\n",
    "uniq_scaf20_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdMag1),uniq_scaf20_df['gRNA_Seq'])\n",
    "\n",
    "print(len(uniq_scaf209_df[uniq_scaf209_df['Match_gRNA']==True]))\n",
    "print(len(uniq_scaf20_df[uniq_scaf20_df['Match_gRNA']==True]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "uniq_scaf209_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdAct1),uniq_scaf209_df['gRNA_Seq_Comp'])\n",
    "uniq_scaf20_df['Match_gRNA'] = map(lambda x: matchSeq(x,hdMag1),uniq_scaf20_df['gRNA_Seq_Comp'])\n",
    "\n",
    "print(len(uniq_scaf209_df[uniq_scaf209_df['Match_gRNA']==True]))\n",
    "print(len(uniq_scaf20_df[uniq_scaf20_df['Match_gRNA']==True]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# uniq_scaf209_df.to_csv()\n",
    "# uniq_scaf20_df.to_csv()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "      <th>gRNA_Seq_Comp</th>\n",
       "      <th>Match_gRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60462</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>3573</td>\n",
       "      <td>3592</td>\n",
       "      <td>GTGAAACGTCGGGATGACGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CACTTTGCAGCCCTACTGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60465</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>6249</td>\n",
       "      <td>6268</td>\n",
       "      <td>GACGGGCACAACACACGGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTGCCCGTGTTGTGTGCCCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60467</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>9278</td>\n",
       "      <td>9297</td>\n",
       "      <td>GTCGGTTTGGTACAGCTGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAGCCAAACCATGTCGACCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60468</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>11741</td>\n",
       "      <td>11760</td>\n",
       "      <td>GCGCCCGATACCCCCGTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGCGGGCTATGGGGGCAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60469</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>12700</td>\n",
       "      <td>12719</td>\n",
       "      <td>GCGCCCGATACTCCCGTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGCGGGCTATGAGGGCAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60470</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>15285</td>\n",
       "      <td>15304</td>\n",
       "      <td>GGCGCACGCCCATTGGTGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCGCGTGCGGGTAACCACCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60480</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>25686</td>\n",
       "      <td>25705</td>\n",
       "      <td>GCAAGCGGAAGATATAACGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGTTCGCCTTCTATATTGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60482</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>28982</td>\n",
       "      <td>29001</td>\n",
       "      <td>GGGTTGAATCACCAAAGTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCCAACTTAGTGGTTTCACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60485</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>30287</td>\n",
       "      <td>30306</td>\n",
       "      <td>GATGCACACAGAATATTTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTACGTGTGTCTTATAAACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60486</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>31605</td>\n",
       "      <td>31624</td>\n",
       "      <td>GCCCTTGAATCAGGTGCAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGGGAACTTAGTCCACGTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60487</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>32129</td>\n",
       "      <td>32148</td>\n",
       "      <td>GCGGAAGAGGGCGTCAGAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGCCTTCTCCCGCAGTCTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60488</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>32132</td>\n",
       "      <td>32151</td>\n",
       "      <td>GAAGAGGGCGTCAGAGGCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTCTCCCGCAGTCTCCGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60489</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>34572</td>\n",
       "      <td>34591</td>\n",
       "      <td>GCCAACACTGTCCTGTCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGGTTGTGACAGGACAGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60497</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>42928</td>\n",
       "      <td>42947</td>\n",
       "      <td>GTATGAAATTGCCAGAGTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CATACTTTAACGGTCTCACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60499</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>44492</td>\n",
       "      <td>44511</td>\n",
       "      <td>GCTTGCTTAGTACCTCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CGAACGAATCATGGAGGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60501</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>44951</td>\n",
       "      <td>44970</td>\n",
       "      <td>GCTTGCTTAGTACCTCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CGAACGAATCATGGAGGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60502</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>45262</td>\n",
       "      <td>45281</td>\n",
       "      <td>GCTTGCTTAGTACCTCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CGAACGAATCATGGAGGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60506</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>53543</td>\n",
       "      <td>53562</td>\n",
       "      <td>GAGGCATCCCCAAATGTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CTCCGTAGGGGTTTACAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60507</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>53842</td>\n",
       "      <td>53861</td>\n",
       "      <td>GAGGCATCCCCAAATGTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CTCCGTAGGGGTTTACAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60513</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>58110</td>\n",
       "      <td>58129</td>\n",
       "      <td>GCTCGCAGAAAAATCCGCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGAGCGTCTTTTTAGGCGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60514</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>58342</td>\n",
       "      <td>58361</td>\n",
       "      <td>GCTCGCAGAAAAGTCCGCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGAGCGTCTTTTCAGGCGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60516</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>60502</td>\n",
       "      <td>60521</td>\n",
       "      <td>GAATCAGAAGATCCGCAAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTAGTCTTCTAGGCGTTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60532</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>67401</td>\n",
       "      <td>67420</td>\n",
       "      <td>GGCGCACGACGTGTGGTGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCGCGTGCTGCACACCACCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60534</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>69799</td>\n",
       "      <td>69818</td>\n",
       "      <td>GGCGATGAGAAGTAAACTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CCGCTACTCTTCATTTGACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60538</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>73496</td>\n",
       "      <td>73515</td>\n",
       "      <td>GGCAGGCACTGACTTTAAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CCGTCCGTGACTGAAATTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60539</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>73795</td>\n",
       "      <td>73814</td>\n",
       "      <td>GGCAGGCACTGACTTTAAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CCGTCCGTGACTGAAATTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60543</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>79241</td>\n",
       "      <td>79260</td>\n",
       "      <td>GAACAGTGGTGCCAGTAAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTGTCACCACGGTCATTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60546</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>79417</td>\n",
       "      <td>79436</td>\n",
       "      <td>GGGATGGTAAGAGGGATGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCCTACCATTCTCCCTACCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60557</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>81044</td>\n",
       "      <td>81063</td>\n",
       "      <td>GAGCGGATGGAGGATAATGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTCGCCTACCTCCTATTACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60558</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>89464</td>\n",
       "      <td>89483</td>\n",
       "      <td>GAACTGCCGAAAGCACTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTGACGGCTTTCGTGAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60561</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>95828</td>\n",
       "      <td>95847</td>\n",
       "      <td>GTGTCTGAATGGAACGCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CACAGACTTACCTTGCGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60562</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>96060</td>\n",
       "      <td>96079</td>\n",
       "      <td>GTGTCTGAATGGAACGCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>+</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CACAGACTTACCTTGCGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60581</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>9182</td>\n",
       "      <td>9163</td>\n",
       "      <td>GGAGATAGCCGGTGAAGCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCTCTATCGGCCACTTCGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60587</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>13091</td>\n",
       "      <td>13072</td>\n",
       "      <td>GGAGTTGATCGCGTCATTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCTCAACTAGCGCAGTAACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60588</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>14186</td>\n",
       "      <td>14167</td>\n",
       "      <td>GTGTATATGCTGCACTGAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CACATATACGACGTGACTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60589</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>14464</td>\n",
       "      <td>14445</td>\n",
       "      <td>GTGTATATGCTGTACTGAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CACATATACGACATGACTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60590</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>14924</td>\n",
       "      <td>14905</td>\n",
       "      <td>GTTTGTGTCTGCGGCGCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAAACACAGACGCCGCGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60595</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>21729</td>\n",
       "      <td>21710</td>\n",
       "      <td>GATGAGCCCGCCGAGGGAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTACTCGGGCGGCTCCCTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60599</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>23924</td>\n",
       "      <td>23905</td>\n",
       "      <td>GCAAAATCGCCGATCCACGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGTTTTAGCGGCTAGGTGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60600</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>26020</td>\n",
       "      <td>26001</td>\n",
       "      <td>GTTTTTCTTCCGCGCAGTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAAAAAGAAGGCGCGTCACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60637</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>67642</td>\n",
       "      <td>67623</td>\n",
       "      <td>GTTTCCCAAAGGTCTGCTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAAAGGGTTTCCAGACGACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60638</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>68777</td>\n",
       "      <td>68758</td>\n",
       "      <td>GATCTCAATTCCATGACAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTAGAGTTAAGGTACTGTCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60640</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>69663</td>\n",
       "      <td>69644</td>\n",
       "      <td>GGTGGTAATAAGGCAAAGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CCACCATTATTCCGTTTCCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60641</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>69693</td>\n",
       "      <td>69674</td>\n",
       "      <td>GGAGGCAGGCCTTGCCGGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCTCCGTCCGGAACGGCCCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60642</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>70401</td>\n",
       "      <td>70382</td>\n",
       "      <td>GCTTTCACCAATACGGTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CGAAAGTGGTTATGCCAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60645</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>72586</td>\n",
       "      <td>72567</td>\n",
       "      <td>GTTTGCAGTCCGTGCACTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAAACGTCAGGCACGTGACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60656</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>73266</td>\n",
       "      <td>73247</td>\n",
       "      <td>GACATCGAAGCACGTTTCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTGTAGCTTCGTGCAAAGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60666</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>85930</td>\n",
       "      <td>85911</td>\n",
       "      <td>GAGGCAGTCCAGATTTTTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTCCGTCAGGTCTAAAAACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60673</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>89718</td>\n",
       "      <td>89699</td>\n",
       "      <td>GGGAGCGTACCAGAATCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CCCTCGCATGGTCTTAGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60674</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>93711</td>\n",
       "      <td>93692</td>\n",
       "      <td>GTTGAGCTACTTAATCCTGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CAACTCGATGAATTAGGACC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60675</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>94375</td>\n",
       "      <td>94356</td>\n",
       "      <td>GATATTTGGGAACAGTCGGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>No</td>\n",
       "      <td>CTATAAACCCTTGTCAGCCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60677</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>96563</td>\n",
       "      <td>96544</td>\n",
       "      <td>GAATATCTGCACCGCCGCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTTATAGACGTGGCGGCGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60678</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>97243</td>\n",
       "      <td>97224</td>\n",
       "      <td>GATGCCGCAGATCTCCCCGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CTACGGCGTCTAGAGGGGCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60680</th>\n",
       "      <td>nHd.2.3.scaf00209</td>\n",
       "      <td>99501</td>\n",
       "      <td>99482</td>\n",
       "      <td>GTAGGAGTAAGCTCCTTAGG</td>\n",
       "      <td>CGG</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>Yes</td>\n",
       "      <td>CATCCTCATTCGAGGAATCC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Contig  Start    End              gRNA_Seq  PAM Strand  \\\n",
       "60462  nHd.2.3.scaf00209   3573   3592  GTGAAACGTCGGGATGACGG  CGG      +   \n",
       "60465  nHd.2.3.scaf00209   6249   6268  GACGGGCACAACACACGGGG  CGG      +   \n",
       "60467  nHd.2.3.scaf00209   9278   9297  GTCGGTTTGGTACAGCTGGG  CGG      +   \n",
       "60468  nHd.2.3.scaf00209  11741  11760  GCGCCCGATACCCCCGTCGG  CGG      +   \n",
       "60469  nHd.2.3.scaf00209  12700  12719  GCGCCCGATACTCCCGTCGG  CGG      +   \n",
       "60470  nHd.2.3.scaf00209  15285  15304  GGCGCACGCCCATTGGTGGG  CGG      +   \n",
       "60480  nHd.2.3.scaf00209  25686  25705  GCAAGCGGAAGATATAACGG  CGG      +   \n",
       "60482  nHd.2.3.scaf00209  28982  29001  GGGTTGAATCACCAAAGTGG  CGG      +   \n",
       "60485  nHd.2.3.scaf00209  30287  30306  GATGCACACAGAATATTTGG  CGG      +   \n",
       "60486  nHd.2.3.scaf00209  31605  31624  GCCCTTGAATCAGGTGCAGG  CGG      +   \n",
       "60487  nHd.2.3.scaf00209  32129  32148  GCGGAAGAGGGCGTCAGAGG  CGG      +   \n",
       "60488  nHd.2.3.scaf00209  32132  32151  GAAGAGGGCGTCAGAGGCGG  CGG      +   \n",
       "60489  nHd.2.3.scaf00209  34572  34591  GCCAACACTGTCCTGTCCGG  CGG      +   \n",
       "60497  nHd.2.3.scaf00209  42928  42947  GTATGAAATTGCCAGAGTGG  CGG      +   \n",
       "60499  nHd.2.3.scaf00209  44492  44511  GCTTGCTTAGTACCTCCCGG  CGG      +   \n",
       "60501  nHd.2.3.scaf00209  44951  44970  GCTTGCTTAGTACCTCCCGG  CGG      +   \n",
       "60502  nHd.2.3.scaf00209  45262  45281  GCTTGCTTAGTACCTCCCGG  CGG      +   \n",
       "60506  nHd.2.3.scaf00209  53543  53562  GAGGCATCCCCAAATGTCGG  CGG      +   \n",
       "60507  nHd.2.3.scaf00209  53842  53861  GAGGCATCCCCAAATGTCGG  CGG      +   \n",
       "60513  nHd.2.3.scaf00209  58110  58129  GCTCGCAGAAAAATCCGCGG  CGG      +   \n",
       "60514  nHd.2.3.scaf00209  58342  58361  GCTCGCAGAAAAGTCCGCGG  CGG      +   \n",
       "60516  nHd.2.3.scaf00209  60502  60521  GAATCAGAAGATCCGCAAGG  CGG      +   \n",
       "60532  nHd.2.3.scaf00209  67401  67420  GGCGCACGACGTGTGGTGGG  CGG      +   \n",
       "60534  nHd.2.3.scaf00209  69799  69818  GGCGATGAGAAGTAAACTGG  CGG      +   \n",
       "60538  nHd.2.3.scaf00209  73496  73515  GGCAGGCACTGACTTTAAGG  CGG      +   \n",
       "60539  nHd.2.3.scaf00209  73795  73814  GGCAGGCACTGACTTTAAGG  CGG      +   \n",
       "60543  nHd.2.3.scaf00209  79241  79260  GAACAGTGGTGCCAGTAAGG  CGG      +   \n",
       "60546  nHd.2.3.scaf00209  79417  79436  GGGATGGTAAGAGGGATGGG  CGG      +   \n",
       "60557  nHd.2.3.scaf00209  81044  81063  GAGCGGATGGAGGATAATGG  CGG      +   \n",
       "60558  nHd.2.3.scaf00209  89464  89483  GAACTGCCGAAAGCACTCGG  CGG      +   \n",
       "60561  nHd.2.3.scaf00209  95828  95847  GTGTCTGAATGGAACGCCGG  CGG      +   \n",
       "60562  nHd.2.3.scaf00209  96060  96079  GTGTCTGAATGGAACGCCGG  CGG      +   \n",
       "60581  nHd.2.3.scaf00209   9182   9163  GGAGATAGCCGGTGAAGCGG  CGG      -   \n",
       "60587  nHd.2.3.scaf00209  13091  13072  GGAGTTGATCGCGTCATTGG  CGG      -   \n",
       "60588  nHd.2.3.scaf00209  14186  14167  GTGTATATGCTGCACTGAGG  CGG      -   \n",
       "60589  nHd.2.3.scaf00209  14464  14445  GTGTATATGCTGTACTGAGG  CGG      -   \n",
       "60590  nHd.2.3.scaf00209  14924  14905  GTTTGTGTCTGCGGCGCCGG  CGG      -   \n",
       "60595  nHd.2.3.scaf00209  21729  21710  GATGAGCCCGCCGAGGGAGG  CGG      -   \n",
       "60599  nHd.2.3.scaf00209  23924  23905  GCAAAATCGCCGATCCACGG  CGG      -   \n",
       "60600  nHd.2.3.scaf00209  26020  26001  GTTTTTCTTCCGCGCAGTGG  CGG      -   \n",
       "60637  nHd.2.3.scaf00209  67642  67623  GTTTCCCAAAGGTCTGCTGG  CGG      -   \n",
       "60638  nHd.2.3.scaf00209  68777  68758  GATCTCAATTCCATGACAGG  CGG      -   \n",
       "60640  nHd.2.3.scaf00209  69663  69644  GGTGGTAATAAGGCAAAGGG  CGG      -   \n",
       "60641  nHd.2.3.scaf00209  69693  69674  GGAGGCAGGCCTTGCCGGGG  CGG      -   \n",
       "60642  nHd.2.3.scaf00209  70401  70382  GCTTTCACCAATACGGTCGG  CGG      -   \n",
       "60645  nHd.2.3.scaf00209  72586  72567  GTTTGCAGTCCGTGCACTGG  CGG      -   \n",
       "60656  nHd.2.3.scaf00209  73266  73247  GACATCGAAGCACGTTTCGG  CGG      -   \n",
       "60666  nHd.2.3.scaf00209  85930  85911  GAGGCAGTCCAGATTTTTGG  CGG      -   \n",
       "60673  nHd.2.3.scaf00209  89718  89699  GGGAGCGTACCAGAATCCGG  CGG      -   \n",
       "60674  nHd.2.3.scaf00209  93711  93692  GTTGAGCTACTTAATCCTGG  CGG      -   \n",
       "60675  nHd.2.3.scaf00209  94375  94356  GATATTTGGGAACAGTCGGG  CGG      -   \n",
       "60677  nHd.2.3.scaf00209  96563  96544  GAATATCTGCACCGCCGCGG  CGG      -   \n",
       "60678  nHd.2.3.scaf00209  97243  97224  GATGCCGCAGATCTCCCCGG  CGG      -   \n",
       "60680  nHd.2.3.scaf00209  99501  99482  GTAGGAGTAAGCTCCTTAGG  CGG      -   \n",
       "\n",
       "      G_start Unique         gRNA_Seq_Comp Match_gRNA  \n",
       "60462    True    Yes  CACTTTGCAGCCCTACTGCC      False  \n",
       "60465    True    Yes  CTGCCCGTGTTGTGTGCCCC      False  \n",
       "60467    True    Yes  CAGCCAAACCATGTCGACCC      False  \n",
       "60468    True    Yes  CGCGGGCTATGGGGGCAGCC      False  \n",
       "60469    True    Yes  CGCGGGCTATGAGGGCAGCC      False  \n",
       "60470    True    Yes  CCGCGTGCGGGTAACCACCC      False  \n",
       "60480    True    Yes  CGTTCGCCTTCTATATTGCC      False  \n",
       "60482    True    Yes  CCCAACTTAGTGGTTTCACC      False  \n",
       "60485    True    Yes  CTACGTGTGTCTTATAAACC      False  \n",
       "60486    True    Yes  CGGGAACTTAGTCCACGTCC      False  \n",
       "60487    True    Yes  CGCCTTCTCCCGCAGTCTCC      False  \n",
       "60488    True    Yes  CTTCTCCCGCAGTCTCCGCC      False  \n",
       "60489    True    Yes  CGGTTGTGACAGGACAGGCC      False  \n",
       "60497    True    Yes  CATACTTTAACGGTCTCACC      False  \n",
       "60499    True     No  CGAACGAATCATGGAGGGCC      False  \n",
       "60501    True     No  CGAACGAATCATGGAGGGCC      False  \n",
       "60502    True     No  CGAACGAATCATGGAGGGCC      False  \n",
       "60506    True     No  CTCCGTAGGGGTTTACAGCC      False  \n",
       "60507    True     No  CTCCGTAGGGGTTTACAGCC      False  \n",
       "60513    True    Yes  CGAGCGTCTTTTTAGGCGCC      False  \n",
       "60514    True    Yes  CGAGCGTCTTTTCAGGCGCC      False  \n",
       "60516    True    Yes  CTTAGTCTTCTAGGCGTTCC      False  \n",
       "60532    True    Yes  CCGCGTGCTGCACACCACCC      False  \n",
       "60534    True     No  CCGCTACTCTTCATTTGACC      False  \n",
       "60538    True     No  CCGTCCGTGACTGAAATTCC      False  \n",
       "60539    True     No  CCGTCCGTGACTGAAATTCC      False  \n",
       "60543    True    Yes  CTTGTCACCACGGTCATTCC      False  \n",
       "60546    True    Yes  CCCTACCATTCTCCCTACCC      False  \n",
       "60557    True    Yes  CTCGCCTACCTCCTATTACC      False  \n",
       "60558    True    Yes  CTTGACGGCTTTCGTGAGCC      False  \n",
       "60561    True     No  CACAGACTTACCTTGCGGCC      False  \n",
       "60562    True     No  CACAGACTTACCTTGCGGCC      False  \n",
       "60581    True    Yes  CCTCTATCGGCCACTTCGCC      False  \n",
       "60587    True    Yes  CCTCAACTAGCGCAGTAACC      False  \n",
       "60588    True     No  CACATATACGACGTGACTCC      False  \n",
       "60589    True     No  CACATATACGACATGACTCC      False  \n",
       "60590    True    Yes  CAAACACAGACGCCGCGGCC      False  \n",
       "60595    True    Yes  CTACTCGGGCGGCTCCCTCC      False  \n",
       "60599    True    Yes  CGTTTTAGCGGCTAGGTGCC      False  \n",
       "60600    True    Yes  CAAAAAGAAGGCGCGTCACC      False  \n",
       "60637    True    Yes  CAAAGGGTTTCCAGACGACC      False  \n",
       "60638    True    Yes  CTAGAGTTAAGGTACTGTCC      False  \n",
       "60640    True     No  CCACCATTATTCCGTTTCCC      False  \n",
       "60641    True    Yes  CCTCCGTCCGGAACGGCCCC      False  \n",
       "60642    True    Yes  CGAAAGTGGTTATGCCAGCC      False  \n",
       "60645    True    Yes  CAAACGTCAGGCACGTGACC      False  \n",
       "60656    True    Yes  CTGTAGCTTCGTGCAAAGCC      False  \n",
       "60666    True    Yes  CTCCGTCAGGTCTAAAAACC      False  \n",
       "60673    True    Yes  CCCTCGCATGGTCTTAGGCC      False  \n",
       "60674    True    Yes  CAACTCGATGAATTAGGACC      False  \n",
       "60675    True     No  CTATAAACCCTTGTCAGCCC      False  \n",
       "60677    True    Yes  CTTATAGACGTGGCGGCGCC      False  \n",
       "60678    True    Yes  CTACGGCGTCTAGAGGGGCC      False  \n",
       "60680    True    Yes  CATCCTCATTCGAGGAATCC      False  "
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scaf209_df[scaf209_df['PAM']=='CGG']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2016-09-12 11:22:33,966 - ngg2 - INFO - ngg2 vv1.3.0\n",
      "\n",
      "Options\n",
      "=======\n",
      "FASTA: nHd.2.3.scaf00209\n",
      "Output file: scaf209_pam_sites.csv\n",
      "Target: All contigs\n",
      "Allow non-canonical starts?: False\n",
      "Max G-bases per site: 15\n",
      "Scan type: Exhaustive\n",
      "Buffered scan: Yes\n",
      "Test site uniqueness: Yes\n",
      "Only unique sites: No\n",
      "Processes: 1\n",
      "\n",
      "2016-09-12 11:22:34,007 - ngg2 - INFO - 223 total sites found by exhaustive scanning\n",
      "2016-09-12 11:22:34,007 - ngg2 - INFO - 170 unique sites (76.2%)\n"
     ]
    }
   ],
   "source": [
    "!ngg2 --outputFile scaf209_pam_sites.csv nHd.2.3.scaf00209\n",
    "# !ngg2 --outputFile scaf209_pam_sites.csv --allowNoncanonical nHd.2.3.scaf00209"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Total Found', 0)\n",
      "('Unique Found', 0)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "      <th>Match_gRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Contig, Start, End, gRNA_Seq, PAM, Strand, G_start, Unique, Match_gRNA]\n",
       "Index: []"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_209 =  pd.read_csv('scaf209_pam_sites.csv')\n",
    "df_209['Match_gRNA'] = map(lambda x: matchSeq(x,hdAct1),df_209['gRNA_Seq'])\n",
    "\n",
    "df_209_u = df_209[df_209.Unique=='Yes']\n",
    "df_209_u['Match_gRNA'] = map(lambda x: matchSeq(x,hdAct1),df_209_u['gRNA_Seq'])\n",
    "\n",
    "print('Total Found',len(df_209[df_209['Match_gRNA']==True]))\n",
    "print('Unique Found',len(df_209_u[df_209_u['Match_gRNA']==True]))\n",
    "\n",
    "\n",
    "df_209_u[df_209_u['Match_gRNA']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2016-09-12 11:22:12,208 - ngg2 - INFO - ngg2 vv1.3.0\n",
      "\n",
      "Options\n",
      "=======\n",
      "FASTA: nHd.2.3.scaf00020\n",
      "Output file: scaf20_pam_sites.csv\n",
      "Target: All contigs\n",
      "Allow non-canonical starts?: True\n",
      "Max G-bases per site: 15\n",
      "Scan type: Exhaustive\n",
      "Buffered scan: Yes\n",
      "Test site uniqueness: Yes\n",
      "Only unique sites: No\n",
      "Processes: 1\n",
      "\n",
      "2016-09-12 11:22:12,323 - ngg2 - INFO - 1877 total sites found by exhaustive scanning\n",
      "2016-09-12 11:22:12,323 - ngg2 - INFO - 1412 unique sites (75.2%)\n"
     ]
    }
   ],
   "source": [
    "# !ngg2 --outputFile scaf20_pam_sites.csv nHd.2.3.scaf00020\n",
    "!ngg2 --outputFile scaf20_pam_sites.csv --allowNoncanonical nHd.2.3.scaf00020"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Total Found', 2)\n",
      "('Unique Found', 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Contig</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gRNA_Seq</th>\n",
       "      <th>PAM</th>\n",
       "      <th>Strand</th>\n",
       "      <th>G_start</th>\n",
       "      <th>Unique</th>\n",
       "      <th>Match_gRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>218115</td>\n",
       "      <td>218134</td>\n",
       "      <td>CGACACGATGATCCGCAAGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>+</td>\n",
       "      <td>False</td>\n",
       "      <td>Yes</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>849</th>\n",
       "      <td>nHd.2.3.scaf00020</td>\n",
       "      <td>218118</td>\n",
       "      <td>218137</td>\n",
       "      <td>CACGATGATCCGCAAGGAGG</td>\n",
       "      <td>TGG</td>\n",
       "      <td>+</td>\n",
       "      <td>False</td>\n",
       "      <td>Yes</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Contig   Start     End              gRNA_Seq  PAM Strand  \\\n",
       "848  nHd.2.3.scaf00020  218115  218134  CGACACGATGATCCGCAAGG  AGG      +   \n",
       "849  nHd.2.3.scaf00020  218118  218137  CACGATGATCCGCAAGGAGG  TGG      +   \n",
       "\n",
       "    G_start Unique Match_gRNA  \n",
       "848   False    Yes       True  \n",
       "849   False    Yes       True  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_20 =  pd.read_csv('scaf20_pam_sites.csv')\n",
    "df_20['Match_gRNA'] = map(lambda x: matchSeq(x,hdMag1),df_20['gRNA_Seq'])\n",
    "\n",
    "df_20_u = df_20[df_20.Unique=='Yes']\n",
    "df_20_u['Match_gRNA'] = map(lambda x: matchSeq(x,hdMag1),df_20_u['gRNA_Seq'])\n",
    "\n",
    "print('Total Found',len(df_20[df_20['Match_gRNA']==True]))\n",
    "print('Unique Found',len(df_20_u[df_20_u['Match_gRNA']==True]))\n",
    "\n",
    "\n",
    "df_20_u[df_20_u['Match_gRNA']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
